src/Archive.php

```diff
@@ -50,12 +50,20 @@ class Archive
 	/**
 	 * Create a new ArchiveStream object.
 	 *
-	 * @param string $name      The name of the resulting archive (optional).
-	 * @param array  $opt       Hash of archive options (see archive options in readme).
-	 * @param string $base_path An optional base path for files to be named under.
+	 * @param resource $output_stream Output stream for archive contents.
+	 * @param string   $name          The name of the resulting archive (optional).
+	 * @param array    $opt           Hash of archive options (see archive options in readme).
+	 * @param string   $base_path     An optional base path for files to be named under.
 	 */
-	public function __construct($name = null, array $opt = array(), $base_path = null)
+	public function __construct(
+		$output_stream,
```

Comment:
This will break backwards compatibility by having this as the first param. Perhaps make it the last with a default of `null` that uses `STDOUT` as its value in the constructor? Purely for the sake of backwards compatibility.

Reply:
Actually, just using `STDOUT` as the default param, like you did later in the PR, should be good after rearranging the order.

Reply:
The only place this constructor gets called though is through the factory and if you look at the `Zip` and `Tar` classes they pass the args through a `func_get_args`

Reply:
And the factory being `instance_by_useragent`

Reply:
Making the change after discussing this offline.